### PR TITLE
AP_TECS.cpp: remove line copy error

### PR DESCRIPTION
**Pull request by rbeall**
This was a copy paste error that is duplicated in the correct spot (in line 530).  This duplicate line effectively resets the height demand too early and model runaway can still occur.

### CODE REVIEW

**Comment by rmackay9** [comment]
I'm not really qualified to approve this but just from looking at the code, this seems correct.  Surely what's in master at the moment is incorrect.

**Comment by magicrub** [comment]
We've been flying this for a while on our fork.

**Comment by IamPete1** [comment]
Looks like this breaks CI, the change looks correct to me so you we may need to update the tests.

**Comment by rbeall-kha** [comment]
> Looks like this breaks CI, the change looks correct to me so we may need to update the tests.

Yeah we’ve added a few more seconds for the climbs on our forked CI. I’ll add it in when I get a chance.

**Comment by tridge** [comment]
this changes looks correct to me, but @priseborough should approve if possible

**Review by priseborough — APPROVED** [review]

**Comment by hendjoshsr71** [comment]
Rebased to kick the tests because they wouldn't rerun

**Comment by rbeall-kha** [comment]
It is in fact failing the RTL climb min CI check, but so far everything seems to be performing exactly as expected.  I think it's just taking longer to climb and therefore taking longer to relax the level roll constraint.  Therefore the timeout on the expectation to make it back to home should be increased.  I'm estimating by how much now.

**Comment by peterbarker** [comment]
autotest changes look reasonable

**Review by priseborough — APPROVED** [review]

**Comment by lopsided98** [comment]
This causes a stair-step pattern in height demand and oscillation in height rate demand during the auto landing descent phase.

image


This log demonstrates the problem: https://drive.google.com/file/d/16j6CjAxdAJdjaBrWJZYLwa5Mgz-4aUk8/view?usp=sharing
This log shows the correct behavior with this PR reverted: https://drive.google.com/file/d/1aQ2bseqsfA3-qkxIBlrF_8_YsATBygI6/view?usp=sharing

**Comment by lopsided98** [comment]
That does seem to be the cause; it looks like my approach was set to be a bit too steep. It doesn't seem like those oscillations are the ideal behavior in that case though.

**Comment by rbeall** [comment]
> That does seem to be the cause; it looks like my approach was set to be a bit too steep. It doesn't seem like those oscillations are the ideal behavior in that case though.

That’s correct. This behavior is expected when climb rate or descent rate commanded is greater than achievable. Reduce glide slope on landing and it should lock in.

**Pull request merged**
